packages/shader-lab/src/codeGen/VisitorContext.ts

```
@@ -52,26 +58,44 @@
-      Logger.error("referenced attribute not found:", ident);
-      return;
+      // #if _EDITOR
+      return new GSError(
+        GSErrorName.CompilationError,
+        `referenced attribute not found: ${ident.lexeme}`,
+        ident.location,
+        ShaderLab._processingPassText
+      );
+      // #else
+      // @ts-ignore
+      throw new Error(`referenced attribute not found: ${ident.lexeme}`);
```

Biome

[error] This code is unreachable

... because this statement will return from the function beforehand


(lint/correctness/noUnreachable)